homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -222,8 +222,8 @@ def device_state_attributes(self):
                 ATTR_FILTER_LEFT: int(
                     self.consumable_state.filter_left.total_seconds()
                     / 3600),
-                ATTR_SENSOR_USED: int(
-                    self.consumable_state.sensor_used.total_seconds()
+                ATTR_SENSOR_DIRTY: int(
+                    self.consumable_state.sensor_dirty.total_seconds()
```

Comment:
I retracted my earlier comment here, but now I'm curious whether this has ever worked? I can't recall it ever being called `sensor_used`.

Reply:
This did not work, it was a mistake on my part. Regarding your previous comment:

I wanted to keep the original functionality and naming from `python-miio`. I have done the math in my config to provide a "left" value.

However, this attribute does work opposite to the rest of them, so I do understand it would probably make more sense.

Reply:
Ok, I saw that you changed it :-) What kind of math you are doing there at the moment? It would be nice if you could just adapt `ConsumableStatus` class in python-miio (it is pretty straightforward, see https://github.com/rytilahti/python-miio/blob/master/miio/vacuumcontainers.py#L284) to provide the also `sensor_dirty_left` (not a really nice name, but follows the suit) and add output for printing it out in the relevant command in `vacuum_cli.py`.

Reply:
Yes, that would be much better. I'll try to get that working.

How many hours until a cleaning of the sensors is recommended (left):
`{{ float(30) - float(states.vacuum.mijia.attributes.sensor_dirty) | round(0) }}`

How many hours passed since the sensors where last cleaned:
`{{ states.vacuum.mijia.attributes.sensor_dirty }}`

Remaining % until a cleaning of the sensors is recommended:
`{{ "%01d" | format(((float(30) - float(states.vacuum.mijia.attributes.sensor_dirty)) / float(30)) * float(100) | int) }}`

The sensors should be cleaned every 30hrs of runtime.

Reply:
Ok, great, thanks for your patch too! I added some comments there earlier, but just for the future reference here's the upstream PR: https://github.com/rytilahti/python-miio/pull/119 .